ui/permissions.py

```diff
@@ -68,4 +204,7 @@ def has_permission(self, request, view):
             uuid.UUID(collection_key)
         except ValueError:
             raise ValidationError('wrong UUID format for {}'.format(collection_key))
-        return Collection.objects.filter(owner=request.user, key=collection_key).exists()
+        collection = Collection.objects.filter(key=collection_key)
+        if collection:
+            return has_admin_permission(collection.first(), request)
+        return False
```

Comment:
The if statement above it will prevent this block from being called if there are no collections (equivalent to False):
```
matches = Collection.objects.filter(title='hfhhf')
if matches:
    print('got a collection')
else:
    print('Nothing there')
```
`> Nothing there`